Improve README

lategoodbye opened the pull request:
Currently the README.md assumed that the reader has also read the AN12 including the product to platform mapping. In order to keep the burden low, replace the product with platform names. This should avoid confusion for new users.

Btw update servicedesk link and drop stray character.

mhei reviewed (changes requested):
The update of the service desk link and the stray character is ok. But I cannot find the product "Tarragon" on the chargebyte's homepage. And in case of EVAcharge SE, it is even worse: this BSP is for Charge Control M only since the partitioning is different from the BSP-only product "EVAcharge SE". So I recommend to stay with the current wording.

lategoodbye commented:
After creating this pull request i noticed the kirkstone-everest branch. I will try to use that for orientation.

lategoodbye committed:
Improve README

Currently the README.md assumed that the reader has also read the
AN12 including the product to platform mapping. So clarify the mapping
between product name and hardware platform.

Btw update servicedesk link and drop stray character.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>

lategoodbye commented:
I force pushed the commit.

mhei reviewed (approved):
Thanks, LGTM now.
